### ISSUE TITLE
Is it reasonable to just reverse the order of training data to create a speech-to-text model

### ISSUE THREAD

**Issue by w013nad** [canopyai/Orpheus-TTS#100]
If I understand everything correctly, is it reasonable and/or a good idea to just reverse the order of the training data? Have the snac tokens be the input and the transcription be the output. Obviously, it would require a retraining of the model but it'd be interesting to see how well the approach would work. In theory, you could use the same pre-tokenized training data.

**Comment by amuvarma13** [comment]
This is actually how end-to-end speech models started off - i.e. check out speechgpt - you don't even need to retrain from scratch - you can training using the base model which already has a good correspondence between text and speech and you should get results quickly. You will need to make sure you only calculate the loss on the text tokens if your input_ids look like speech-text otherwise your model will still focus on learning modelling speech tokens using the previous speech tokens. 

You should get okay results - there are models that already work like this - they won't be SOTA like whisper, but from an academic perspective it could be quite interesting,
